x/bank/types/key.go

```diff
@@ -43,15 +44,24 @@ func AddressFromBalancesStore(key []byte) (sdk.AccAddress, error) {
 	if len(key) == 0 {
 		return nil, ErrInvalidKey
 	}
+
 	addrLen := key[0]
 	bound := int(addrLen)
+
 	if len(key)-1 < bound {
 		return nil, ErrInvalidKey
 	}
+
 	return key[1 : bound+1], nil
 }
 
 // CreateAccountBalancesPrefix creates the prefix for an account's balances.
 func CreateAccountBalancesPrefix(addr []byte) []byte {
 	return append(BalancesPrefix, address.MustLengthPrefix(addr)...)
 }
+
+// CreateDenomPrefix creates a prefix for a reverse index of denomination to
+// account balance for that denomination.
+func CreateDenomPrefix(denom string) []byte {
+	return append(DenomPrefix, []byte(denom)...)
```

Comment:
Actually we need to null terminate this string because neither addresses nor denoms are fixed length.

Reply:
Instead of null-terminating the denom component, we could also `LengthPrefix` it: `{denom_length} ++ {denom}`

Reply:
Right either way will work

Reply:
The downside of length prefixing is that denoms with the same prefix will not be grouped together.

Reply:
Do you have an example of a prefix/index that is null-terminating?